src/core/database/database.service.ts

```diff
@@ -170,6 +175,55 @@ export class DatabaseService {
     });
   }
 
+  // Used for user, education and unavailability, when a user wants to
+  // update their own properties.
+  async updatePropertiesInsecure<TObject extends Resource>({
```

Comment:
Can we just add an option to skip authorization checks on `updateProperties` instead of duplicating all of this?

```tsx
updateProperties({
  ...
  skipAuth: true,
});
```